diff --git a/td/generate/tl_writer_dotnet.h b/td/generate/tl_writer_dotnet.h
index b6eac7a2d..000000000 100644
--- a/td/generate/tl_writer_dotnet.h
+++ b/td/generate/tl_writer_dotnet.h
@@ -138,6 +138,9 @@ class TlWriterDotNet final : public TL_writer {
   }
 
   std::string gen_class_name(std::string name) const final {
+    if (name == "linkPreview") {
+      return "WebPage";
+    }
     if (name == "Object" || name == "#") {
       assert(0);
     }
@@ -145,6 +148,9 @@ class TlWriterDotNet final : public TL_writer {
   }
 
   std::string gen_field_name(std::string name) const final {
+    if (name == "link_preview") {
+      return "WebPage";
+    }
     assert(name.size() > 0);
     assert(is_alnum(name.back()));
     return to_CamelCase(name);
@@ -370,7 +376,20 @@ class TlWriterDotNet final : public TL_writer {
   std::string gen_additional_function(const std::string &function_name, const tl_combinator *t,
                                       bool is_function) const final {
     std::stringstream ss;
+    auto class_name = gen_class_name(t->name);
     if (is_header_ && function_name == "ToUnmanaged") {
+      if (class_name == "WebPage") {
+        ss << "  property ::Telegram::Td::Api::Animation^ Animation { ::Telegram::Td::Api::Animation^ get() { return nullptr; } }\n"
+              "  property ::Telegram::Td::Api::Audio^ Audio { ::Telegram::Td::Api::Audio^ get() { return nullptr; } }\n"
+              "  property ::Telegram::Td::Api::Document^ Document { ::Telegram::Td::Api::Document^ get() { return nullptr; } }\n"
+              "  property ::Telegram::Td::Api::Photo^ Photo { ::Telegram::Td::Api::Photo^ get() { return nullptr; } }\n"
+              "  property ::Telegram::Td::Api::Sticker^ Sticker { ::Telegram::Td::Api::Sticker^ get() { return nullptr; } }\n"
+              "  property ::Telegram::Td::Api::Video^ Video { ::Telegram::Td::Api::Video^ get() { return nullptr; } }\n"
+              "  property ::Telegram::Td::Api::VideoNote^ VideoNote { ::Telegram::Td::Api::VideoNote^ get() { return nullptr; } }\n"
+              "  property ::Telegram::Td::Api::VoiceNote^ VoiceNote { ::Telegram::Td::Api::VoiceNote^ get() { return nullptr; } }\n"
+              "  property int32 EmbedWidth { int32 get() { return 0; } }\n"
+              "  property int32 EmbedHeight { int32 get() { return 0; } }\n";
+      }
       ss << "};\n";
     }
     ss << "\n";
